### PR TITLE
Fix for LRN Layer

### LINKED ISSUE #3360: [Bug] LRN layer is broken for CPU
The `LRN` layer is broken for CPU but not GPU. The following crashes:

```
import mxnet as mx
s = mx.symbol.LRN(nsize=4, data=mx.symbol.Variable("data"))
ctx = mx.cpu()
executor = s.simple_bind(ctx=ctx, data=(1,3,255,255))
executor.forward(is_train=False)
```

and generates the following output to stderr:

```
[17:16:12] /Users/sebastianb/Software/mxnet/dmlc-core/include/dmlc/logging.h:235: [17:16:12] src/operator/./lrn-inl.h:59: Check failed: (out_data.size()) == (2)
[17:16:12] /Users/sebastianb/Software/mxnet/dmlc-core/include/dmlc/logging.h:235: [17:16:12] src/engine/./threaded_engine.h:306: [17:16:12] src/operator/./lrn-inl.h:59: Check failed: (out_data.size()) == (2)
An fatal error occurred in asynchronous engine operation. If you do not know what caused this error, you can try set environment variable MXNET_ENGINE_TYPE to NaiveEngine and run with debugger (i.e. gdb). This will force all operations to be synchronous and backtrace will give you the series of calls that lead to this error. Remember to set MXNET_ENGINE_TYPE back to empty after debugging.
libc++abi.dylib: terminating with uncaught exception of type dmlc::Error: [17:16:12] src/engine/./threaded_engine.h:306: [17:16:12] src/operator/./lrn-inl.h:59: Check failed: (out_data.size()) == (2)
```

### PR DESCRIPTION
**Pull request by sbodenstein**
This PR fixes #3360 in the simplest way (and not the most efficient: extra backward dependencies are required for the cuDNN and mshadow version. Efficiency can improved later).

@piiswrong

### CODE REVIEW

File: src/operator/lrn-inl.h [hunk 1]
```diff
@@ -140,43 +138,35 @@ class LocalResponseNormProp : public OperatorProperty {
...
     const std::vector<int> &in_data,
     const std::vector<int> &out_data) const override {
-#if MXNET_USE_CUDNN == 1
-    return {out_grad[lrn_enum::kOut], in_data[lrn_enum::kData], out_data[lrn_enum::kOut]};
-#else
-    return {out_grad[lrn_enum::kOut], in_data[lrn_enum::kData], out_data[lrn_enum::kTmpNorm]};
-#endif
+    return {
+      out_grad[lrn_enum::kOut], in_data[lrn_enum::kData],
+      out_data[lrn_enum::kTmpNorm], out_data[lrn_enum::kTmpNorm],
```
Comment (piiswrong) [inline src/operator/lrn-inl.h]: duplicated ktmpnorm?

Comment (sbodenstein) [inline src/operator/lrn-inl.h]: oh dear, sorry about that. Just fixed it.

File: src/operator/lrn-inl.h [hunk 2]
```diff
@@ -140,43 +138,34 @@ class LocalResponseNormProp : public OperatorProperty {
...
 
   std::vector<std::pair<int, void*> > BackwardInplaceOption(
     const std::vector<int> &out_grad,
     const std::vector<int> &in_data,
     const std::vector<int> &out_data,
     const std::vector<void*> &in_grad) const override {
-#if MXNET_USE_CUDNN == 1
-    return {};
-#else
     return {{out_grad[lrn_enum::kOut], in_grad[lrn_enum::kData]}};
```
Comment (piiswrong) [inline src/operator/lrn-inl.h]: This breaks cudnn because cudnn doesn't support inplace

Comment (sbodenstein) [inline src/operator/lrn-inl.h]: Fixed.

**Pull request merged**